feature_engine/imputation/categorical.py

```diff
@@ -175,6 +185,10 @@ def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
             # select all variables or check variables entered by the user
             self.variables_ = find_all_variables(X, self.variables)
 
+        # identify variables with missing values
+        if self.missing_only and self.variables is None:
```

Comment:
same as previous, we need to combine this with find_all_variables so it doesnt search twice.

Reply:
See my prior response. By implementing the elif approach, we'll encounter a similar issue. This time we only want categorical variables, but if we create a new elif then we'll have both categorical and numerical variables